manim/mobject/geometry/labeled.py

```diff
@@ -2,17 +2,102 @@
 
 from __future__ import annotations
 
-__all__ = ["LabeledLine", "LabeledArrow"]
+__all__ = ["Label", "LabeledLine", "LabeledArrow", "LabeledPolygram"]
+
+import numpy as np
 
 from manim.constants import *
 from manim.mobject.geometry.line import Arrow, Line
+from manim.mobject.geometry.polygram import Polygram
 from manim.mobject.geometry.shape_matchers import (
     BackgroundRectangle,
     SurroundingRectangle,
 )
 from manim.mobject.text.tex_mobject import MathTex, Tex
 from manim.mobject.text.text_mobject import Text
+from manim.mobject.types.vectorized_mobject import VGroup
 from manim.utils.color import WHITE, ManimColor, ParsableManimColor
+from manim.utils.polylabel import PolyLabel
+
+
+class Label(VGroup):
+    """A Label consisting of text and frame
+
+    Parameters
+    ----------
+    label : str | Tex | MathTex | Text
+        Label that will be displayed on the line.
+    font_size : float | optional
+        Control font size for the label. This parameter is only used when `label` is of type `str`.
+    label_color: ParsableManimColor | optional
+        The color of the label's text. This parameter is only used when `label` is of type `str`.
+    label_frame : Bool | optional
+        Add a `SurroundingRectangle` frame to the label box.
+    frame_fill_color : ParsableManimColor | optional
+        Background color to fill the label box. If no value is provided, the background color of the canvas will be used.
+    frame_fill_opacity : float | optional
+        Determine the opacity of the label box by passing a value in the range [0-1], where 0 indicates complete transparency and 1 means full opacity.
+
+    Examples
+    --------
+    .. manim:: LabelExample
+        :save_last_frame:
+        :quality: high
+
+        class LabelExample(Scene):
+            def construct(self):
+                label = Label(
+                    label          = Text('Label Text', font='sans-serif'),
+                    label_color    = WHITE,
+                    label_frame    = True
+                )
+                label.scale(3)
+                self.add(label)
+    """
+
+    def __init__(
+        self,
+        label: str | Tex | MathTex | Text,
+        font_size: float = DEFAULT_FONT_SIZE,
+        label_color: ParsableManimColor = WHITE,
+        label_frame: bool = True,
+        frame_fill_color: ParsableManimColor = None,
+        frame_fill_opacity: float = 1,
+        **kwargs,
+    ) -> None:
```

Comment:
It is not necessary to write again the parameter types inside the docstring. Only the signature typings are needed.

Maybe the `label_frame` parameter could be deleted, because IMO it doesn't make sense to use a `Label` without adding a frame to it. In that case, one could simply use `Tex` or `Text`. If it does make sense, then I would suggest a name which better reflects its "booleanness", such as `use_label_frame` or `add_label_frame`.

It would be great if people was able to modify text and frame attributes other than the label's font size or color, or the frame's fill color or opacity. Many other `Mobject`s in Manim are composed of other submobjects which can be configured independently via `config` dict parameters. Therefore, I'd suggest the same approach and replace these parameters by `label_config` and `frame_config` dictionaries.

```suggestion
class Label(VGroup):
    """A Label consisting of a text surrounded by a frame.

    Parameters
    ----------
    label
        Label that will be displayed on the line.
    label_config
        A dictionary containing the configuration for the label. This is only applied if `label` is of type `str`.
    frame_config
        A dictionary containing the configuration for the frame.
        
    Examples
    --------
    .. manim:: LabelExample
        :save_last_frame:
        :quality: high
        class LabelExample(Scene):
            def construct(self):
                label = Label(
                    label=Text('Label Text', font='sans-serif', color=WHITE),
                )
                label.scale(3)
                self.add(label)
    """

    def __init__(
        self,
        label: str | Tex | MathTex | Text,
        label_config: dict[str, Any] | None = None,
        frame_config: dict[str, Any] | None = None,
        **kwargs,
    ) -> None:
```